LightBlue.MultiHost/Configuration/CustomRunnerConfiguration.cs

```
@@ -0,0 +1,10 @@
+namespace LightBlue.MultiHost.Configuration
+{
+    public class CustomRunnerConfiguration
+    {
+        public string Name { get; set; }
+        public string Command { get; set; }
+        public string Arguments { get; set; } = string.Empty;
+        public string Icon { get; set; } = "website";
```

I'm not clear how this string is used, but the default Icon for our services is worker (https://github.com/LightBlueProject/LightBlue/blob/2ab46db6b961cfc7ef44e26132f73110e81ba809/LightBlue.MultiHost/ViewModel/Role.cs#L82)

---

Before that switch statement we do 
```CSharp
 if (App.Configuration.CustomRunners.FirstOrDefault(x => x.Name == Config.RoleName) is CustomRunnerConfiguration customRunner)
                    return $"Resources\\{customRunner.Icon}.ico";
```

---

Ah thanks, I see how this is hanging together now.
Few things to consider;
1. Custom runners have a default icon of `website`, all other runners have a default icon of `worker`. I think we should default to `worker` here as well for consistent behaviour.
2. We should separate the configuration values from the internal resource names (if we decide to change our internal icon resource names it shouldn't make everyone's config incompatible) - perhaps an enum for the Configuration's Icon parameter that we map later to the resource name?
3. What is the intended user experience if they get the icon config parameter wrong? Multihost crash? Error message for the specific role? Silent failure and default icon?

---

1 & 2: I have updated the default and created a new IconOption enum which will be used in the config and then converted into the resource name via a helper class.

3: Originally it would just show a blank icon but it will now just use the default (worker) if theres no match